Makefile

```diff
@@ -393,7 +394,19 @@ uninstall:
 	  rm -f $(DESTDIR)$(docdir)/`basename $$f`; \
 	done
 
-.PHONY: installdirs install-program install-data install uninstall
+installcheck:
+	@rm -rf testinstall || true
+	$(MAKE) DESTDIR=testinstall install
+	$(VALGRIND) $(VALGRIND_TEST_ARGS) testinstall$(bindir)/lightningd --test-daemons-only --lightning-dir=testinstall
```

Comment:
Even better, set --lightning-dir to testinstall/lightningdir and the checks will sure it *doesn't* exist after this...